src/language-html/printer-html.js

```diff
@@ -63,7 +64,42 @@ function concat(parts) {
 
 function embed(path, print, textToDoc, options) {
   const node = path.getValue();
+
   switch (node.type) {
+    case "element": {
+      if (isScriptLikeTag(node) || node.type === "interpolation") {
+        // Fall through to "text"
+        return;
+      }
+
+      if (isVueNonHtmlBlock(node, options)) {
+        const parser = inferScriptParser(node, options);
+        if (!parser) {
+          return;
+        }
+
+        let doc;
+        try {
+          doc = textToDoc(getNodeContent(node, options), { parser });
```

Comment:
This text pass to parser need discuss too, 

```js
<custom lang="pug">
  div
</custom>
```

What we should pass to plugin? 
- `"\n  div\n"` (current)
- `"  div"` (trim leading and trailing empty lines)
- `"  div\n"` (trim ltrailing empty lines)
- ~`"div"` (this should not consider)~

Reply:
I would prefer the exact behavior as currently with the embedded markdown

Reply:
Which is?

if same

````
```pug
  div
```
````

Reply:
Seems removed one line on each side, means extra blank lines kept. https://github.com/prettier/prettier/blob/master/src/language-markdown/utils.js#L220

Reply:
_Beside the fact that my plugin has a bug_

_embedded markdown_

````
```pug
  div
```
````

**currently** formats to (bug)

````
```pug
| div
```
````

but **should** format to 

````
```pug
div
```
````